lambdas/build-user-identity/src/test/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandlerTest.java

```diff
@@ -517,11 +302,19 @@ void shouldReturnCredentialsWithCiMitVCOnSuccessfulUserInfoRequest() throws Exce
         AuditExtensionsUserIdentity extensions =
                 (AuditExtensionsUserIdentity) capturedAuditEvent.getExtensions();
         assertEquals(VectorOfTrust.P0.toString(), extensions.getLevelOfConfidence());
-        assertTrue(extensions.isCiFail());
+        assertFalse(extensions.isCiFail());
         assertFalse(extensions.isHasMitigations());
         assertEquals(extensions.getReturnCodes().size(), 3);
-        assertTrue(extensions.getReturnCodes().stream().anyMatch(rt -> rt.code().equals("3")));
-        assertFalse(extensions.getReturnCodes().stream().allMatch(rt -> rt.issuers().isEmpty()));
+        assertEquals(extensions.getReturnCodes().get(0).code(), "1");
+        assertEquals(extensions.getReturnCodes().get(0).issuers(), Collections.emptyList());
+        assertEquals(extensions.getReturnCodes().get(1).code(), "2");
+        assertEquals(
+                extensions.getReturnCodes().get(1).issuers(),
+                List.of("https://review-q.account.gov.uk", "https://review-f.account.gov.uk"));
+        assertEquals(extensions.getReturnCodes().get(2).code(), "3");
+        assertEquals(
+                extensions.getReturnCodes().get(2).issuers(),
+                List.of("https://review-z.account.gov.uk", "https://review-f.account.gov.uk"));
         verify(mockClientOAuthSessionDetailsService, times(1)).getClientOAuthSession(any());
         verify(mockCiMitService, times(1)).getContraIndicatorsVCJwt(any(), any(), any());
 
```

Comment:
I'm not entirely sure the test below (line 325) is needed. The test above checks the code path where the return code isn't found in the CI config map already. The test below is just an extreme example of that, one that wouldn't actually happen?

Reply:
Ok, I deleted this test.